### PR TITLE
fix: Session replay transformed view masking

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Fixing some masking regression and added test to track it.

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: Sources/Swift/Tools/SentryViewPhotographer.swift [hunk 1]
```diff
@@ -45,6 +45,9 @@ class SentryViewPhotographer: NSObject, SentryViewScreenshotProvider {
         dispatchQueue.dispatchAsync {
             let screenshot = UIGraphicsImageRenderer(size: imageSize, format: .init(for: .init(displayScale: 1))).image { context in
                 
+                let clipOutPath = CGMutablePath(rect: CGRect(origin: .zero, size: imageSize), transform: nil)
```
Comment (philipphofmann) [inline Sources/Swift/Tools/SentryViewPhotographer.swift:48]: `m`: I see no new tests for these changes here, nor does `testRedactLabelWithParentTransformed` fail if I revert the changes here. What made you make those changes? These changes don't like a refactoring, if I'm not mistaken, and it would be great to see some updates to tests if they change behavior.

Comment (brustolin) [reply Sources/Swift/Tools/SentryViewPhotographer.swift:48]: Ow, true. Forgot a test for this change.

File: Tests/SentryTests/SentryViewPhotographerTests.swift [hunk 2]
```diff
@@ -144,6 +144,22 @@ class SentryViewPhotographerTests: XCTestCase {
...
+        parentView.backgroundColor = .green
+        parentView.transform = CGAffineTransform(rotationAngle: 90 * .pi / 180.0)
+        parentView.addSubview(label)
+        
+        let image = try XCTUnwrap(prepare(views: [parentView] ))
+        let pixel1 = color(at: CGPoint(x: 10, y: 10), in: image)
+        assertColor(pixel1, .white)
+        
+        let pixel2 = color(at: CGPoint(x: 22, y: 10), in: image)
+        assertColor(pixel2, .black)
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Tests/SentryTests/SentryViewPhotographerTests.swift:160]: > As long as the general region is masked

The test only checks two pixels. How can you be sure the whole region is masked correctly in this scenario?

> and we have other tests to check the rectangle that is being masked

Also, when using `transform`?

[the remark this answers is not in the document]
Comment (brustolin) [reply Tests/SentryTests/SentryViewPhotographerTests.swift:160]: I guess we could add another test with transformation in `UIRedactBuilderTests` that is responsible for the calculation of the region.

[the remark this answers is not in the document]
Comment (brustolin) [reply Tests/SentryTests/SentryViewPhotographerTests.swift]: I decided to test in the ViewPhotographer because transformation, especially rotation, needs to work with long float numbers, and dealing with floating imprecision can cause flakiness.

So, it will be easier to test with the final image.

File: Tests/SentryTests/SentryViewPhotographerTests.swift [hunk 3]
```diff
@@ -210,6 +237,44 @@ class SentryViewPhotographerTests: XCTestCase {
...
+        let topView = UIView(frame: CGRect(x: 25, y: 0, width: 25, height: 25))
+        topView.backgroundColor = .green
+        
+        let label1 = UILabel(frame: CGRect(x: 0, y: 0, width: 50, height: 25))
+        label1.text = "Test"
+        label1.textColor = .black
+        
+        let parentView = UIView(frame: CGRect(x: 0, y: 0, width: 50, height: 25))
+        parentView.addSubview(label1)
+        parentView.clipsToBounds = true
```
Comment (philipphofmann) [inline Tests/SentryTests/SentryViewPhotographerTests.swift:250]: `m`: If we want to test `clipsToBounds`, shouldn't the `label1` be bigger than the `parentView`. Currently, they are the same size, and `clipsToBounds` shouldn't make a difference. Or is there another reason to use it?

I think if you set the background color of the parentView to red and increase the height of the label1 to 30, no red color should be visible, cause the label and the topView should be drawn over the parentView.

File: Tests/SentryTests/SentryViewPhotographerTests.swift [hunk 4]
```diff
@@ -210,6 +237,44 @@ class SentryViewPhotographerTests: XCTestCase {
...
+        
+        let image = try XCTUnwrap(prepare(views: [parentView, topView]))
```
Comment (philipphofmann) [inline Tests/SentryTests/SentryViewPhotographerTests.swift:252]: `m`: It would be highly beneficial to store the image to a temp location for debugging purposes or when a test fails to understand better what's going on with something like:
```
if let data = image.pngData() {
    let url = FileManager.default.temporaryDirectory.appendingPathComponent("testImage.png")
    try? data.write(to: url)
    print("Image saved to: \(url.path)")
}
```

File: Tests/SentryTests/SentryViewPhotographerTests.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Tests/SentryTests/SentryViewPhotographerTests.swift]: For testing readability, you could use the exact expected values and create an assert helper method that allows some deviation.

File: Tests/SentryTests/SentryViewPhotographerTests.swift (hunk 3 above)
Comment (brustolin) [reply Tests/SentryTests/SentryViewPhotographerTests.swift:250]: Not for this test. I just need to parent to clip its bounds because it changes the flow of masking.

File: Tests/SentryTests/SentryViewPhotographerTests.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Tests/SentryTests/SentryViewPhotographerTests.swift]: Maybe in another PR. My feedback is that it's hard to read these tests. You have to mentally map the coordinates to check if it's working correctly.

**Review by philipphofmann — APPROVED** [review]
LGTM
